src/client/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -129,6 +129,31 @@ public async Task<IAccount> GetAccountAsync(string accountId)
             return accounts.FirstOrDefault(account => account.HomeAccountId.Identifier.Equals(accountId, StringComparison.OrdinalIgnoreCase));
         }
 
+        /// <summary>
+        /// Get the <see cref="IAccount"/> by its identifier among the accounts available in the token cache,
+        /// based on the user flow. This is for Azure AD B2C scenarios.
+        /// </summary>
+        /// <param name="userFlow">User flow identifier. The identifier is the user flow being targeted by the specific B2C authority/>.
+        /// </param>
+        public async Task<IAccount> GetAccountByUserFlowAsync(string userFlow)
```

Comment:
We don't have the "UserFlow" concept in MSALs as of now. Can we get agreement from other MSALs on introducing a new concept in the library? Or was this discussed and approved already by the B2C working group?

I am also unsure if filtering by objectId is a robust mechanism - was this discussed with server side folks?

Reply:
we were told by b2c team to use "userFlow" in regards to the policy. this is to reduce the load on the developer, as msal should be handling this. When we designed the cache, B2C was overlooked, so we pushed this onto the developer. I don't want this in identity web. I cad hide it behind a feature flag if necessary.

> I am also unsure if filtering by objectId is a robust mechanism - was this discussed with server side folks?

Then why are we forcing customers to rely on this to filter the cache?

Reply:
btw...msal .net doesn't have a concept of `KnownAuthorities` but other libs do, like obj-c and js.

Reply:
KnownAuthorities is a known and accepted concept which all libraries are supposed to implement as part of sdk-consistency.

The current implementation makes the GetAccounts method is internal, didn't you mean to make it public?

We should discuss whether this functionality should be added to MSAL or not.